arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1092,7 +1119,40 @@ void ClusterInfo::loadPlan() {
     // we can safely keep this iterator around because we hold the read-lock on _planProt here.
     // reusing the lookup positions helps avoiding redundant lookups into _plannedCollections
     // for the same database
-    AllCollections::const_iterator existingCollections = _plannedCollections.find(databaseName);
+
+    AllCollections::const_iterator existingCollections,
+      stillExistingCollections = newCollections.find(databaseName);
+    {
+      READ_LOCKER(guard, _planProt.lock);
+      existingCollections = _plannedCollections.find(databaseName);
+    }
+
+    if (stillExistingCollections != newCollections.end()) {
+      auto const& np = newPlan.find(databaseName);
+      if (np != newPlan.end()) {
+        auto nps = np->second->slice()[0];
+        for (auto const& ec : *(stillExistingCollections->second)) {
+          auto const& cid = ec.first;
+          if (!std::isdigit(cid.front())) {
```

Comment:
Should there be an additional check for the emptyness of `cid`?

From https://en.cppreference.com/w/cpp/string/basic_string/front:

> Returns reference to the first character in the string. The behavior is undefined if empty() == true.
```suggestion
          if (cid.empty() || !std::isdigit(cid.front())) {
```